woocommerce/woocommerce-gutenberg-products-block
# Product Gallery Thumbnails: Fix overflow issues and improve responsiveness

- This PR attempts to improve the way Thumbnails display and re-size in the editor and the frontend, preventing them from overflowing the container.
- This PR also changes the minimum number of thumbnails from `2` to `3`.

## What

Fixes woocommerce/woocommerce-blocks#11014

## Why

Currently, the thumbnails have a hardcoded size and overflow the container.

## Important notes

1. The Thumbnails cropping needs to be fixed in a separate PR:
- Slack reference: p1697717112780469/1697706500.523259-slack-C02FL3X7KR6
- https://github.com/woocommerce/woocommerce-blocks/pull/11718
2. This PR **does not** fix the thumbnails styling in the dialog. That will need to be completely separate.


## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._


1. With a block theme, open `Appearance > Editor`.
3. Open `Templates > Single Product`.
4. Select `Convert to Blocks` if you are using the Classic Templates.
5. Add the `Product Gallery` block.
6. Ensure the Thumbnails are enabled.
7. Test how thumbnails are displaying for various `Number of Thumbnails` values.
8. Test with thumbnails `Position` set to `off`, `left`, `bottom` and `top`.
9. Test both the editor and front end.


## Screenshots or screencast



### Editor

| Before | After |
| ------ | ----- |
|<img width="907" alt="before_editor" src="https://github.com/woocommerce/woocommerce-blocks/assets/905781/7f9f3cb3-aa23-47bb-b2b2-a450db1e59b5">|<img width="1235" alt="after_editor" src="https://github.com/woocommerce/woocommerce-blocks/assets/905781/0b7beb78-7955-48bf-8532-d1c20b2a1f5b">|

### Front end

| Before | After |
| ------ | ----- |
|<img width="1066" alt="before__frontend" src="https://github.com/woocommerce/woocommerce-blocks/assets/905781/4c7c533c-f95e-4dac-9510-a099ddde9042">|<img width="1083" alt="after_frontend" src="https://github.com/woocommerce/woocommerce-blocks/assets/905781/9dbda094-5d89-417b-87d8-701f9f99786d">|

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Product Gallery Thumbnails: Fix overflow issues and improve responsiveness.
